client/src/hooks/permissions/index.ts

```diff
@@ -0,0 +1,16 @@
+import { useProfile } from 'hooks/profile';
+
+import type { Permission } from './enums';
+
+export function usePermissions() {
+  const { data, isLoading } = useProfile();
+  const roles = data?.roles[0];
```

Comment:
Take in consideration that a user might have multiple roles with multiple permissions attached to each role

Reply:
yes, roles should be an array